src/software/ai/hl/stp/tactic/tactic.h

```diff
@@ -90,7 +137,33 @@ class Tactic
      *
      * @return the next primitive
      */
-    std::map<RobotId, std::shared_ptr<Primitive>> get(const WorldPtr &world_ptr);
+    std::map<RobotId, std::shared_ptr<Primitive>> get(const WorldPtr &world_ptr)
+    {
+        TbotsProto::RobotNavigationObstacleConfig obstacle_config;
+        std::map<RobotId, std::shared_ptr<Primitive>> primitives_map;
+
+        {
+            ZoneNamedN(_tracy_tactic_set_primitive,
+                       "Tactic: Get primitives for each robot", true);
+
+            for (const auto &robot : world_ptr->friendlyTeam().getAllRobots())
+            {
+                updatePrimitive(
+                    TacticUpdate(robot, world_ptr,
+                                 [this](std::shared_ptr<Primitive> new_primitive)
+                                 { primitive = std::move(new_primitive); }),
+                    !last_execution_robot.has_value() ||
+                        last_execution_robot.value() != robot.id());
+
+                CHECK(primitive != nullptr)
+                    << "Primitive for " << objectTypeName(*this) << " in state "
+                    << getFSMState() << " was not set" << std::endl;
+                primitives_map[robot.id()] = std::move(primitive);
+            }
+        }
+
+        return primitives_map;
+    }
```

Comment:
I'd move the method implementations outside of the class declaration (unless it's a very short method, like a getter), just so that we keep a similar code style to our other template class files (e.g. [this one](https://github.com/UBC-Thunderbots/Software/blob/da152fa10729ae4aac2023c577c7e68205e290d9/src/software/networking/unix/proto_unix_listener.hpp))